ckcp/gitops.sh

```diff
@@ -0,0 +1,340 @@
+        set -x
+        ;;
+      -h|--help)
+        usage
+        exit 0
+        ;;
+      --)
+        # End of arguments
+        break
+      ;;
+      *)
+        echo "Unknown argument: $1"
+        usage
+        exit 1
+        ;;
+    esac
+    shift
+  done
+}
+
+
+
+check_cluster_role(){
+  if [ "$(oc auth can-i '*' '*' --all-namespaces)" != "yes" ]; then
+    echo
+    echo "[ERROR] User '$(oc whoami)' does not have the required 'cluster-admin' role." 1>&2
+    echo "Log into the cluster with a user with the required privileges (e.g. kubeadmin) and retry."
+    exit 1
+  fi
+}
+
+
+
+install_openshift_gitops(){
+  APP="openshift-gitops"
+  export KUBECONFIG="$KUBECONFIG_CLUSTER"
+
+  local ns="openshift-operators"
+
+  #############################################################################
+  # Install the gitops operator
+  #############################################################################
+  echo -n "  - OpenShift-GitOps: "
+  if ! oc get subscriptions -n "$ns" openshift-gitops-operator >/dev/null 2>&1; then
+    oc apply -f "$GITOPS_DIR/$APP.yaml" --wait >/dev/null
+  fi
+  echo "OK"
+
+  #############################################################################
+  # Wait for the URL to be available
+  #############################################################################
+  echo -n "  - ArgoCD dashboard: "
+  test_cmd="oc get route/openshift-gitops-server --ignore-not-found -n $APP -o jsonpath={.spec.host}"
+  ARGOCD_HOSTNAME="$(${test_cmd[@]})"
+  until curl --fail --insecure --output /dev/null --silent "https://$ARGOCD_HOSTNAME"; do
+    echo -n "."
+    sleep 2
+    ARGOCD_HOSTNAME="$(${test_cmd[@]})"
+  done
+  echo "OK"
+  echo "  - ArgoCD URL: https://$ARGOCD_HOSTNAME"
+
+  #############################################################################
+  # Post install
+  #############################################################################
+
+  # Allow ArgoCD to control resources outside of its namespace
+  oc adm policy add-cluster-role-to-user cluster-admin -z openshift-gitops-argocd-application-controller -n openshift-gitops >/dev/null
+
+  # Log into ArgoCD
+  echo -n "  - ArgoCD Login: "
+  local argocd_password
+  argocd_password="$(oc get secret openshift-gitops-cluster -n openshift-gitops -o jsonpath="{.data.admin\.password}" | base64 --decode --ignore-garbage)"
+  argocd login "$ARGOCD_HOSTNAME" --grpc-web --insecure --username admin --password "$argocd_password" >/dev/null
+  echo "OK"
+}
+
+
+
+install_ckcp(){
+  APP="ckcp"
+  export KUBECONFIG="$KUBECONFIG_CLUSTER"
+
+  local ns="$APP"
+  local kube_dir="$SCRIPT_DIR/kubeconfig"
+  local test_cmd
+
+  #############################################################################
+  # Deploy KCP
+  #############################################################################
+  echo -n "  - $APP application: "
+  if ! oc get apps -n openshift-gitops "$APP" >/dev/null 2>&1; then
+    touch "$kube_dir/.gitignore"
```

Comment:
This is to know when was the last time we created a new KCP cluster.

Later on we do
```
  # Copy the kubeconfig of kcp from inside the pod onto the local filesystem
  if [ "$kube_dir/.gitignore" -nt "$KUBECONFIG_KCP" ]; then
    oc cp ""$APP"/$podname:/workspace/.kcp/admin.kubeconfig" "$KUBECONFIG_KCP" >/dev/null
  fi
```

1. I'm trying to avoid having to copy, and the sed the file for the external_ip on every run
2. But I do want the file to be regenerated if for example the ckcp namespace was deleted and a new ckcp deployment was done.